modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -1174,10 +1176,11 @@ private void handleSortByClause(final Query query) {
                 }
 
                 // assign to query if sorting is supported!
-                final SortBy[] sb = clauses.toArray(new SortBy[] {});
+
+                this.sortBy = clauses.toArray(new SortBy[] {});
                 if (rasterManager.getGranuleCatalog()
-                        .getQueryCapabilities(rasterManager.getTypeName()).supportsSorting(sb)) {
-                    query.setSortBy(sb);
+                        .getQueryCapabilities(rasterManager.getTypeName()).supportsSorting(sortBy)) {
+                    query.setSortBy(sortBy);
```

Comment:
I think we're going to leave this for the user to control with the SortBy parameter for now. In the future I'd like to update maybe SubmosaicProducerFactory to have an optional SortBy parameter so that the RasterLayerRequest can query.